### DISCUSSION TITLE
Is Viem suitable for backend Node.js apps?

### DISCUSSION THREAD

**Discussion by AlissonRS** [wevm/viem#1651] [Question] [answered]
I was going through the migration guide to ethers.js, and noticed that for BigInt, Viem uses the **browser's** builtin BigInt API, so does that mean Viem is not suitable for backend apps (where there is no browser)?

I haven't tried using it yet, just thought I would ask in case someone already knows the answer.

Thanks!

**Comment by jxom** [comment, marked as answer]
`BigInt` is supported on Node: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#browser_compatibility

Plenty of folks are using Viem in a Node backend!

**Reply by AlissonRS** [reply]
Thanks @jxom !
 
 I'm considering switching from ethers.js + typechain to viem + ABIType + wagmi (as we also have Next.js apps).
 
 So this would require refactoring our contract unit tests, backend (Node.js) and frontend (Next.js).
 
 From the smart contracts repo, I publish an NPM package with to be used in backend and frontend backend apps, it's based on typechain + ethers.js, so I'd also change our NPM packages to export ABIType, and let client apps (backend and frontend) decide whether they want to use Viem or Ethers.js (I'm assuming ABIType types are compatible with both).
